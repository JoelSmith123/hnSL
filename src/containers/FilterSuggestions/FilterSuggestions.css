.FilterSuggestions {
  align-items: center;
  background-color: #fff;
  display: flex;
  flex-direction: column;
  height: calc(100vh - 50px);
  justify-content: center;
  width: 100%;  
}

.filter-title {
  font-family: 'Roboto Condensed', sans-serif;
  font-weight: bold;
  font-size: 3rem;
}

.filter-input-categories-container {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 900px;
}

.filter-name-input-title {
  color: #fff;
  font-family: 'Roboto Condensed', sans-serif;
}

.filter-name-input-title-bool {
  color: #fff;
  font-family: 'Roboto Condensed', sans-serif;
  margin-bottom: 80px;
}

.filter-input-form-container {
  background-color: #171636; 
  box-shadow: 0px 1px 10px rgba(0, 0, 0, 0.25);
  box-sizing: border-box;
  border-radius: 15px;
  display: flex;
  margin: 0 auto;
  margin-top: 30px;
  width: 400px;
}

.filter-input-form {
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  padding-left: 40px;
  width: 400px;
}

.filter-input {
  border: 3px solid #6981FE;
  border-radius: 20px;
  box-sizing: border-box;
  font-family: 'Roboto Condensed', sans-serif;
  font-size: 1rem;
  height: 60px;
  margin-bottom: 20px;
  padding-left: 10px;
  width: 240px;
}

.add-input-btn {
  margin-bottom: 20px;
  margin-left: 20px;
  margin-right: 30px;
  width: 60px;  
}

.filter-btn,
.input-btn,
.add-input-btn,
.category-btn {
  transition: background-color 200ms;
}

.filter-btn:hover,
.input-btn:hover,
.add-input-btn:hover,
.category-btn:hover {
  background-color: #a5b3fe;
}

.filter-btn:active,
.input-btn:active,
.add-input-btn:active,
.category-btn:active {
  background-color: #34407f;
}


.input-btn {
  align-self: flex-end;
  background-color: #6981FE;
  border: none;
  border-radius: 20px;
  box-sizing: border-box;
  color: #fff;
  flex-basis: 100%;
  font-family: Roboto Condensed;
  font-weight: bold;
  font-size: 1.5rem;
  height: 60px;
}

.filter-categories-input-title {
  grid-column: 1 / 4;
}

.input-categories-form {
  display: grid;
  grid-template-rows: 60px 1fr 1fr;
  grid-template-columns: 1fr 1fr 1fr;
  justify-items: center;
  width: 100%;
}

.category-btn {
  align-self: center;
  margin: 10px;
  width: 90px;
}


.filter-btn {
  background-color: #6981FE;
  border: none;
  border-radius: 20px;
  box-sizing: border-box;
  color: #fff;
  font-family: Roboto Condensed;
  font-weight: bold;
  font-size: 1.5rem;
  height: 60px;
  margin-top: 40px;
  width: 200px;
}


.input-btn-unactive {
  background-color: #808080;
  border: none;
  border-radius: 20px;
  box-sizing: border-box;
  color: #fff;
  flex-basis: 100%;
  font-family: Roboto Condensed;
  font-weight: bold;
  font-size: 1.5rem;
  height: 60px; 
}


