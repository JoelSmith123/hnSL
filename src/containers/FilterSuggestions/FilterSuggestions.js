import React, { Component } from 'react'
import './FilterSuggestions.css'

export default class FilterSuggestions extends Component {
  render() {
    return (
      <div className='FilterSuggestions'>
        <h1 className='filter-title'>More filters</h1>
        <form>
          <input className='filter-input-1' placeholder='Enter title...'></input>
          <input className='filter-input-2' placeholder='Enter title...'></input>
          <button className='filter-btn'>Find suggestions</button>
        </form>
      </div>
    )
  }
}