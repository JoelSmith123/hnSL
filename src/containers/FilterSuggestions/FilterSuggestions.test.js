import { FilterSuggestions, mapDispatchToProps } from './FilterSuggestions.js'
import { setFilter, setCategory } from '../../actions/index.js'
import React from 'react'
import { shallow } from 'enzyme'

describe('FilterSuggestions', () => {
  const wrapper = shallow(<FilterSuggestions setFilter={jest.fn()} setCategory={jest.fn()}/>)

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('handleChange', () => {
    it('should update local state with input values', () => {
      const mockEvent = { target: {name: 'input-0',  value: 'underoath'} }
      const mockState = { inputs: {'input-0': 'underoath'} }

      wrapper.instance().handleChange(mockEvent)

      expect(wrapper.state().inputs).toEqual(mockState.inputs)
    })      
  })

  describe('handleSubmit', () => {
    it('should call setFilter on form submit', () => {
      const mockEvent = { target: {}, preventDefault: () => {} }

      wrapper.instance().handleSubmit(mockEvent)

      expect(wrapper.instance().props.setFilter).toHaveBeenCalled()
    })

    it('should call setCategory on form submit', () => {
      const mockEvent = { target: {}, preventDefault: () => {} }

      wrapper.instance().handleSubmit(mockEvent)

      expect(wrapper.instance().props.setCategory).toHaveBeenCalled()
    })

    it('should update local state on form submit', () => {
      const mockEvent = { target: {}, preventDefault: () => {} }

      wrapper.instance().handleSubmit(mockEvent)

      expect(wrapper.state()).toEqual({activeBtn: '', category: '', inputs: {}})
    })    
  })

  describe('handleClick', () => {
    it('should update local state on category button click', () => {
      const mockEvent = { target: {name: 'games'}, preventDefault: () => {} }

      wrapper.instance().handleClick(mockEvent)

      expect(wrapper.state()).toEqual({category: 'games', activeBtn: 'games', inputs: {}})
    })
  })

  describe('appendInput', () => {
    it('should update state with new inputs', () => {
      const mockEvent = { target: {}, preventDefault: () => {} }
      
      wrapper.setState({inputs: {'input-0': ''}})

      wrapper.instance().appendInput(mockEvent)

      expect(wrapper.state().inputs).toEqual({'input-0': '', 'input-1': ''})
    })
  })

})

describe('mapDispatchToProps', () => {
  it('calls dispatch with a setFilter action when setFilter is called', () => {
    const mockDispatch = jest.fn()
    const actionToDispatch = setFilter({'input-0': ''})

    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.setFilter({'input-0': ''})

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

  it('calls dispatch with a setCategory action when setCategory is called', () => {
    const mockDispatch = jest.fn()
    const actionToDispatch = setCategory('games')

    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.setCategory('games')

    expect(mockDispatch).toBeCalledWith(actionToDispatch)
  })
})