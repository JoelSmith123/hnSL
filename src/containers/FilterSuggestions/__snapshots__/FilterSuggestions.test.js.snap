// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterSuggestions matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FilterSuggestions
    setCategory={[MockFunction]}
    setFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="filter-title"
        >
          Filters
        </h1>,
        null,
        <div
          className="filter-input-categories-container"
        >
          <div
            className="filter-input-form-container"
          >
            <form
              className="filter-input-form"
            >
              <h3
                className="filter-name-input-title-bool"
              >
                Enter a game, movie, band, or song
              </h3>
              <input
                className="filter-input"
                name="input-0"
                onChange={[Function]}
                placeholder=""
                value=""
              />
            </form>
            <button
              className="add-input-btn input-btn"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              +
            </button>
          </div>
          <div
            className="filter-input-form-container"
          >
            <form
              className="input-categories-form"
              onClick={[Function]}
            >
              <h3
                className="filter-name-input-title filter-categories-input-title"
              >
                Select a category
              </h3>
              <button
                className="category-btn input-btn"
                name="games"
                onClick={[Function]}
                type="button"
              >
                games
              </button>
              <button
                className="category-btn input-btn"
                name="music"
                onClick={[Function]}
                type="button"
              >
                music
              </button>
              <button
                className="category-btn input-btn"
                name="authors"
                onClick={[Function]}
                type="button"
              >
                authors
              </button>
              <button
                className="category-btn input-btn"
                name="movies"
                onClick={[Function]}
                type="button"
              >
                movies
              </button>
              <button
                className="category-btn input-btn"
                name="shows"
                onClick={[Function]}
                type="button"
              >
                shows
              </button>
              <button
                className="category-btn input-btn"
                name="books"
                onClick={[Function]}
                type="button"
              >
                books
              </button>
            </form>
          </div>
        </div>,
        <button
          className="filter-btn"
          onClick={[Function]}
          type="submit"
        >
          Find suggestions
        </button>,
      ],
      "className": "FilterSuggestions",
      "onKeyPress": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Filters",
          "className": "filter-title",
        },
        "ref": null,
        "rendered": "Filters",
        "type": "h1",
      },
      null,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="filter-input-form-container"
            >
              <form
                className="filter-input-form"
              >
                <h3
                  className="filter-name-input-title-bool"
                >
                  Enter a game, movie, band, or song
                </h3>
                <input
                  className="filter-input"
                  name="input-0"
                  onChange={[Function]}
                  placeholder=""
                  value=""
                />
              </form>
              <button
                className="add-input-btn input-btn"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                +
              </button>
            </div>,
            <div
              className="filter-input-form-container"
            >
              <form
                className="input-categories-form"
                onClick={[Function]}
              >
                <h3
                  className="filter-name-input-title filter-categories-input-title"
                >
                  Select a category
                </h3>
                <button
                  className="category-btn input-btn"
                  name="games"
                  onClick={[Function]}
                  type="button"
                >
                  games
                </button>
                <button
                  className="category-btn input-btn"
                  name="music"
                  onClick={[Function]}
                  type="button"
                >
                  music
                </button>
                <button
                  className="category-btn input-btn"
                  name="authors"
                  onClick={[Function]}
                  type="button"
                >
                  authors
                </button>
                <button
                  className="category-btn input-btn"
                  name="movies"
                  onClick={[Function]}
                  type="button"
                >
                  movies
                </button>
                <button
                  className="category-btn input-btn"
                  name="shows"
                  onClick={[Function]}
                  type="button"
                >
                  shows
                </button>
                <button
                  className="category-btn input-btn"
                  name="books"
                  onClick={[Function]}
                  type="button"
                >
                  books
                </button>
              </form>
            </div>,
          ],
          "className": "filter-input-categories-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <form
                  className="filter-input-form"
                >
                  <h3
                    className="filter-name-input-title-bool"
                  >
                    Enter a game, movie, band, or song
                  </h3>
                  <input
                    className="filter-input"
                    name="input-0"
                    onChange={[Function]}
                    placeholder=""
                    value=""
                  />
                </form>,
                <button
                  className="add-input-btn input-btn"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  +
                </button>,
              ],
              "className": "filter-input-form-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h3
                      className="filter-name-input-title-bool"
                    >
                      Enter a game, movie, band, or song
                    </h3>,
                    Array [
                      <input
                        className="filter-input"
                        name="input-0"
                        onChange={[Function]}
                        placeholder=""
                        value=""
                      />,
                    ],
                  ],
                  "className": "filter-input-form",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Enter a game, movie, band, or song",
                      "className": "filter-name-input-title-bool",
                    },
                    "ref": null,
                    "rendered": "Enter a game, movie, band, or song",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": "input-0",
                    "nodeType": "host",
                    "props": Object {
                      "className": "filter-input",
                      "name": "input-0",
                      "onChange": [Function],
                      "placeholder": "",
                      "value": "",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "form",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "+",
                  "className": "add-input-btn input-btn",
                  "disabled": false,
                  "onClick": [Function],
                  "type": "button",
                },
                "ref": null,
                "rendered": "+",
                "type": "button",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                className="input-categories-form"
                onClick={[Function]}
              >
                <h3
                  className="filter-name-input-title filter-categories-input-title"
                >
                  Select a category
                </h3>
                <button
                  className="category-btn input-btn"
                  name="games"
                  onClick={[Function]}
                  type="button"
                >
                  games
                </button>
                <button
                  className="category-btn input-btn"
                  name="music"
                  onClick={[Function]}
                  type="button"
                >
                  music
                </button>
                <button
                  className="category-btn input-btn"
                  name="authors"
                  onClick={[Function]}
                  type="button"
                >
                  authors
                </button>
                <button
                  className="category-btn input-btn"
                  name="movies"
                  onClick={[Function]}
                  type="button"
                >
                  movies
                </button>
                <button
                  className="category-btn input-btn"
                  name="shows"
                  onClick={[Function]}
                  type="button"
                >
                  shows
                </button>
                <button
                  className="category-btn input-btn"
                  name="books"
                  onClick={[Function]}
                  type="button"
                >
                  books
                </button>
              </form>,
              "className": "filter-input-form-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h3
                    className="filter-name-input-title filter-categories-input-title"
                  >
                    Select a category
                  </h3>,
                  <button
                    className="category-btn input-btn"
                    name="games"
                    onClick={[Function]}
                    type="button"
                  >
                    games
                  </button>,
                  <button
                    className="category-btn input-btn"
                    name="music"
                    onClick={[Function]}
                    type="button"
                  >
                    music
                  </button>,
                  <button
                    className="category-btn input-btn"
                    name="authors"
                    onClick={[Function]}
                    type="button"
                  >
                    authors
                  </button>,
                  <button
                    className="category-btn input-btn"
                    name="movies"
                    onClick={[Function]}
                    type="button"
                  >
                    movies
                  </button>,
                  <button
                    className="category-btn input-btn"
                    name="shows"
                    onClick={[Function]}
                    type="button"
                  >
                    shows
                  </button>,
                  <button
                    className="category-btn input-btn"
                    name="books"
                    onClick={[Function]}
                    type="button"
                  >
                    books
                  </button>,
                ],
                "className": "input-categories-form",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Select a category",
                    "className": "filter-name-input-title filter-categories-input-title",
                  },
                  "ref": null,
                  "rendered": "Select a category",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "games",
                    "className": "category-btn input-btn",
                    "name": "games",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "games",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "music",
                    "className": "category-btn input-btn",
                    "name": "music",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "music",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "authors",
                    "className": "category-btn input-btn",
                    "name": "authors",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "authors",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "movies",
                    "className": "category-btn input-btn",
                    "name": "movies",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "movies",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "shows",
                    "className": "category-btn input-btn",
                    "name": "shows",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "shows",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "books",
                    "className": "category-btn input-btn",
                    "name": "books",
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "books",
                  "type": "button",
                },
              ],
              "type": "form",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Find suggestions",
          "className": "filter-btn",
          "onClick": [Function],
          "type": "submit",
        },
        "ref": null,
        "rendered": "Find suggestions",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="filter-title"
          >
            Filters
          </h1>,
          null,
          <div
            className="filter-input-categories-container"
          >
            <div
              className="filter-input-form-container"
            >
              <form
                className="filter-input-form"
              >
                <h3
                  className="filter-name-input-title-bool"
                >
                  Enter a game, movie, band, or song
                </h3>
                <input
                  className="filter-input"
                  name="input-0"
                  onChange={[Function]}
                  placeholder=""
                  value=""
                />
              </form>
              <button
                className="add-input-btn input-btn"
                disabled={false}
                onClick={[Function]}
                type="button"
              >
                +
              </button>
            </div>
            <div
              className="filter-input-form-container"
            >
              <form
                className="input-categories-form"
                onClick={[Function]}
              >
                <h3
                  className="filter-name-input-title filter-categories-input-title"
                >
                  Select a category
                </h3>
                <button
                  className="category-btn input-btn"
                  name="games"
                  onClick={[Function]}
                  type="button"
                >
                  games
                </button>
                <button
                  className="category-btn input-btn"
                  name="music"
                  onClick={[Function]}
                  type="button"
                >
                  music
                </button>
                <button
                  className="category-btn input-btn"
                  name="authors"
                  onClick={[Function]}
                  type="button"
                >
                  authors
                </button>
                <button
                  className="category-btn input-btn"
                  name="movies"
                  onClick={[Function]}
                  type="button"
                >
                  movies
                </button>
                <button
                  className="category-btn input-btn"
                  name="shows"
                  onClick={[Function]}
                  type="button"
                >
                  shows
                </button>
                <button
                  className="category-btn input-btn"
                  name="books"
                  onClick={[Function]}
                  type="button"
                >
                  books
                </button>
              </form>
            </div>
          </div>,
          <button
            className="filter-btn"
            onClick={[Function]}
            type="submit"
          >
            Find suggestions
          </button>,
        ],
        "className": "FilterSuggestions",
        "onKeyPress": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Filters",
            "className": "filter-title",
          },
          "ref": null,
          "rendered": "Filters",
          "type": "h1",
        },
        null,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="filter-input-form-container"
              >
                <form
                  className="filter-input-form"
                >
                  <h3
                    className="filter-name-input-title-bool"
                  >
                    Enter a game, movie, band, or song
                  </h3>
                  <input
                    className="filter-input"
                    name="input-0"
                    onChange={[Function]}
                    placeholder=""
                    value=""
                  />
                </form>
                <button
                  className="add-input-btn input-btn"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  +
                </button>
              </div>,
              <div
                className="filter-input-form-container"
              >
                <form
                  className="input-categories-form"
                  onClick={[Function]}
                >
                  <h3
                    className="filter-name-input-title filter-categories-input-title"
                  >
                    Select a category
                  </h3>
                  <button
                    className="category-btn input-btn"
                    name="games"
                    onClick={[Function]}
                    type="button"
                  >
                    games
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="music"
                    onClick={[Function]}
                    type="button"
                  >
                    music
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="authors"
                    onClick={[Function]}
                    type="button"
                  >
                    authors
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="movies"
                    onClick={[Function]}
                    type="button"
                  >
                    movies
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="shows"
                    onClick={[Function]}
                    type="button"
                  >
                    shows
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="books"
                    onClick={[Function]}
                    type="button"
                  >
                    books
                  </button>
                </form>
              </div>,
            ],
            "className": "filter-input-categories-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <form
                    className="filter-input-form"
                  >
                    <h3
                      className="filter-name-input-title-bool"
                    >
                      Enter a game, movie, band, or song
                    </h3>
                    <input
                      className="filter-input"
                      name="input-0"
                      onChange={[Function]}
                      placeholder=""
                      value=""
                    />
                  </form>,
                  <button
                    className="add-input-btn input-btn"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    +
                  </button>,
                ],
                "className": "filter-input-form-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <h3
                        className="filter-name-input-title-bool"
                      >
                        Enter a game, movie, band, or song
                      </h3>,
                      Array [
                        <input
                          className="filter-input"
                          name="input-0"
                          onChange={[Function]}
                          placeholder=""
                          value=""
                        />,
                      ],
                    ],
                    "className": "filter-input-form",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "Enter a game, movie, band, or song",
                        "className": "filter-name-input-title-bool",
                      },
                      "ref": null,
                      "rendered": "Enter a game, movie, band, or song",
                      "type": "h3",
                    },
                    Object {
                      "instance": null,
                      "key": "input-0",
                      "nodeType": "host",
                      "props": Object {
                        "className": "filter-input",
                        "name": "input-0",
                        "onChange": [Function],
                        "placeholder": "",
                        "value": "",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                  ],
                  "type": "form",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "+",
                    "className": "add-input-btn input-btn",
                    "disabled": false,
                    "onClick": [Function],
                    "type": "button",
                  },
                  "ref": null,
                  "rendered": "+",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  className="input-categories-form"
                  onClick={[Function]}
                >
                  <h3
                    className="filter-name-input-title filter-categories-input-title"
                  >
                    Select a category
                  </h3>
                  <button
                    className="category-btn input-btn"
                    name="games"
                    onClick={[Function]}
                    type="button"
                  >
                    games
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="music"
                    onClick={[Function]}
                    type="button"
                  >
                    music
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="authors"
                    onClick={[Function]}
                    type="button"
                  >
                    authors
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="movies"
                    onClick={[Function]}
                    type="button"
                  >
                    movies
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="shows"
                    onClick={[Function]}
                    type="button"
                  >
                    shows
                  </button>
                  <button
                    className="category-btn input-btn"
                    name="books"
                    onClick={[Function]}
                    type="button"
                  >
                    books
                  </button>
                </form>,
                "className": "filter-input-form-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h3
                      className="filter-name-input-title filter-categories-input-title"
                    >
                      Select a category
                    </h3>,
                    <button
                      className="category-btn input-btn"
                      name="games"
                      onClick={[Function]}
                      type="button"
                    >
                      games
                    </button>,
                    <button
                      className="category-btn input-btn"
                      name="music"
                      onClick={[Function]}
                      type="button"
                    >
                      music
                    </button>,
                    <button
                      className="category-btn input-btn"
                      name="authors"
                      onClick={[Function]}
                      type="button"
                    >
                      authors
                    </button>,
                    <button
                      className="category-btn input-btn"
                      name="movies"
                      onClick={[Function]}
                      type="button"
                    >
                      movies
                    </button>,
                    <button
                      className="category-btn input-btn"
                      name="shows"
                      onClick={[Function]}
                      type="button"
                    >
                      shows
                    </button>,
                    <button
                      className="category-btn input-btn"
                      name="books"
                      onClick={[Function]}
                      type="button"
                    >
                      books
                    </button>,
                  ],
                  "className": "input-categories-form",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Select a category",
                      "className": "filter-name-input-title filter-categories-input-title",
                    },
                    "ref": null,
                    "rendered": "Select a category",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "games",
                      "className": "category-btn input-btn",
                      "name": "games",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "games",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "music",
                      "className": "category-btn input-btn",
                      "name": "music",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "music",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "authors",
                      "className": "category-btn input-btn",
                      "name": "authors",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "authors",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "movies",
                      "className": "category-btn input-btn",
                      "name": "movies",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "movies",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "shows",
                      "className": "category-btn input-btn",
                      "name": "shows",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "shows",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "books",
                      "className": "category-btn input-btn",
                      "name": "books",
                      "onClick": [Function],
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": "books",
                    "type": "button",
                  },
                ],
                "type": "form",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Find suggestions",
            "className": "filter-btn",
            "onClick": [Function],
            "type": "submit",
          },
          "ref": null,
          "rendered": "Find suggestions",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
